packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -492,20 +491,33 @@ class _CupertinoAlertDialogState extends State<CupertinoAlertDialog> {
   }
 }
 
-/// Rounded rectangle surface that looks like an iOS popup surface, e.g., alert dialog
-/// and action sheet.
+/// An iOS-style rounded rectangle surface useful for showing modal overlays,
+/// such as dialogs and action sheets.
 ///
-/// A [CupertinoPopupSurface] can be configured to paint or not paint a white
-/// color on top of its blurred area. Typical usage should paint white on top
-/// of the blur. However, the white paint can be disabled for the purpose of
-/// rendering divider gaps for a more complicated layout, e.g., [CupertinoAlertDialog].
-/// Additionally, the white paint can be disabled to render a blurred rounded
-/// rectangle without any color (similar to iOS's volume control popup).
+/// By default, this surface is composed of a transparent surface color painted
+/// atop a [BackdropFilter]. The surface color is light gray when the ambient
+/// [CupertinoTheme] brightness is [Brightness.light], and dark gray when the
+/// ambient brightness is [Brightness.dark]. The [BackdropFilter] applied
+/// beneath the surface saturates and blurs underlying content. In other words,
+/// the overall effect is result = add_surface_color(blur(saturate(source))).
```

Comment:
Sorry to bikeshed again, but I'm thinking of explaining it without technical terms, such as `BackdropFilter`.

```suggestion
/// An iOS-style component for creating modal overlays like dialogs and action sheets.
///
/// By default, [CupertinoPopupSurface] generates a rounded rectangle surface that applies two effects
/// to the background content:
///
///   1. Background filter: Increases saturation and then adds blur to the content behind
///      the surface.
///   2. Overlay color: Covers the filtered background with a transparent surface color.
///      The color adapts to the CupertinoTheme's brightness: Light gray when 
///      [CupertinoTheme] brightness is [Brightness.light], and dark gray when [Brightness.dark].
```

Reply:
No worries -- it does read better. The only thing I changed was 
`Increases saturation and then adds blur to the content behind the surface` =>  `Saturates and then blurs content behind the surface`, but let me know if you want me to change it back.

Reply:
That's definitely better!